src/context/index.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { computed, IArrayWillSplice, intercept, observable, observe } from 'mobx'
+import { Log, Result } from 'sarif'
+import { commands, DiagnosticSeverity, ExtensionContext, languages, Memento, Range, Selection, TextDocument, ThemeColor, Uri, window, workspace } from 'vscode'
+import { mapDistinct, _Region, parseRegion } from '../shared'
+import '../shared/extension'
+import { Baser } from './Baser'
+import { loadLogs } from './loadLogs'
+import { Panel } from './Panel'
+
+declare module 'vscode' {
+	interface Diagnostic {
+		_result: Result
+	}
+}
+
+export const regionToSelection = (doc: TextDocument, region: _Region | undefined) => {
```

Comment:
>export const regionToSelection =  [](start = 0, length = 33)

Why this syntax versus exporting a function?

Reply:
I nearly never want the `this` semantic of `function`. So I stick with `arrow functions`. This is true even when the `this` semantic isn't significant just to keep simple.

You're welcome to criticize me for `function activate` a few lines below which is a carryover from the sample code which this all started from. I'm not being self-consistent here and I should. :-)